aspnetcore/mvc/controllers/filters.md

```diff
@@ -86,216 +82,224 @@ Several of the filter interfaces have corresponding attributes that can be used
 Filter attributes:
 
 * `ActionFilterAttribute`
-
 * `ExceptionFilterAttribute`
-
 * `ResultFilterAttribute`
-
 * `FormatFilterAttribute`
-
 * `ServiceFilterAttribute`
-
 * `TypeFilterAttribute`
 
-### Cancellation and Short Circuiting
+`TypeFilterAttribute` and `ServiceFilterAttribute` are explained [later in this article](#dependency-injection).
 
-You can short-circuit the filter pipeline at any point by setting the `Result` property on the context parameter provided to the filter method. For instance, the following `ShortCircuitingResourceFilter` will prevent any other filters from running later in the pipeline, including any action filters.
+## Filter scopes
 
-<a name=short-circuiting-resource-filter></a>
+A filter can be added to the pipeline at one of three *scopes*. You can add a filter to a particular action method or to a controller class by using an attribute. Or you can register a filter globally by adding it to the `MvcOptions.Filters` collection in the `ConfigureServices` method in the `Startup` class:
```

Comment:
It's not really suitable to give this information without explaining how scopes determine order of execution.